### DISCUSSION TITLE
Orchid - Adjust queries to Database

### DISCUSSION THREAD

**Discussion by CarlsonBuma** [orchidsoftware/platform#1687] [Q&A]
Hello, I'm kinda new in Laravel - Orchid.
So I'm not very familiar with the Framework and I couldnt found it in my Project nor in the internet research.

So I try to learn about the structure of Orchid, belonging to query building to CRUD in Database.
I currently using it as Backend Framework (admin).

I need to change a query - cause I receive a ForeignKey but I want to have the relationship to the PK. 
How to Search for Table Relations?

![image](https://user-images.githubusercontent.com/75643791/115846521-49dd1100-a422-11eb-8c7c-44337c68a81a.png)

Would also be nice, if you could include examples in your Doc, which shows the structure of your Integration & query building.

Btw. its a very nice platform! Thanks alot

![image](https://user-images.githubusercontent.com/75643791/115858102-d4c40880-a42e-11eb-958a-f85b74590177.png)

**Comment by kubre** [comment]
Hello @CarlsonBuma ,
From what I understand you want to show Author name or other fields from authors table in Author column instead showing the foreign key.

In your PostListScreen which has query method you should eager load authors(this will reduce amount of queries)
```php
public function query(): array
{
   return [
        'posts' => Post::filters()
            ->with('author') # You must define this relation on Post model
            ->paginate(),
    ];
}
``` 

Now in your PostListLayout:
```php
class PostsListLayout extends Table
{

    protected $target = 'admissions';

    protected function columns(): array
    {
        return [
            # Accessing fields on a relation with 'relation_name.attribute_name' 
            TD::make('author.name', 'Auhtor'),

            # OR alternatively
            TD::make('author_name', 'Auhtor')
              ->(function(Post $post) { return $post->author->name; }), 
         ];
    }
}
```
Both work fine, though I prefer first syntax as its simple
Hope this helps.

**Reply by CarlsonBuma** [reply]
@kubre The Post model, I need to process as documented - somewhere - in **Laravel "Eloquent: Relationships"**? I cant figure out how to create a relation to Parent. But thank you in advance for your help. Very helpful so far!
![image](https://user-images.githubusercontent.com/75643791/115941519-f90efc00-a4a5-11eb-885f-608d9df41322.png)
* Updated pic. sorry

**Reply by kubre** [reply]
Though its not related to orchid but just laravel in general here is the documentation from laravel website

[Defining Belongs To relationship in laravel](https://laravel.com/docs/8.x/eloquent-relationships#one-to-one-defining-the-inverse-of-the-relationship)

**Reply by CarlsonBuma** [reply]
ok thanks anyway 👍

**Comment by mrbig00** [comment]
Any idea how to apply the filters and sorting for this?
